src/django_components_js/build.py

```diff
@@ -0,0 +1,54 @@
+import os
+import subprocess
+from pathlib import Path
+from typing import List, Optional, Sequence, Union
+
+DEFAULT_ESBUILD_ARGS = [
+    "--bundle",
+    "--minify",
+    # "--sourcemap",  # NOTE: Enable for debugging during development
+    "--platform=browser",
+    "--target=chrome80,firefox73,safari13,edge80",
+]
+
+
+# Compile a list of JS/TS files into a single minified file with esbuild
+def compile_js_files_to_file(
+    file_paths: Sequence[Union[Path, str]],
+    out_file: Union[Path, str],
+    esbuild_args: Optional[List[str]] = None,
+):
+    # Find Esbuild binary
+    bin_name = "esbuild.cmd" if os.name == "nt" else "esbuild"
+    esbuild_path = Path(os.getcwd()) / "node_modules" / ".bin" / bin_name
+
+    # E.g. `esbuild js_file1.ts js_file2.ts js_file3.ts --bundle --minify --outfile=here.js`
+    esbuild_cmd = [
+        str(esbuild_path),
+        *[str(filepath) for filepath in file_paths],
+        *(esbuild_args if esbuild_args is not None else DEFAULT_ESBUILD_ARGS),
+        # Compile into a single file
+        f"--outfile={out_file}",
+    ]
+
+    # check=True should ensure that this raises an error if the subprocess fails.
+    subprocess.run(esbuild_cmd, check=True)
+
+
+# NOTE:
+# - This script should be called from within django_components_js` dir!
+# - Also you need to have esbuild installed. If not yet, run:
+#   `npm install -D esbuild`
+def build():
+    entrypoint = "./src/index.ts"
+    out_file = Path("../django_components/static/django_components/django_components.min.js")
+
+    # Prepare output dir
+    os.makedirs(out_file.parent, exist_ok=True)
+
+    # Compile JS
+    compile_js_files_to_file(file_paths=[entrypoint], out_file=out_file)
+
+
+if __name__ == "__main__":
+    build()
```

Comment:
@EmilStenstrom Let me know if this is fine with you.

So following our convo in the tech doc, I moved the JS tests into Playwright (See the [`test_dependency_manager.py`](https://github.com/EmilStenstrom/django-components/pull/666/files#diff-1734962c259079fe44ab215607c037d799acd31b8f29f8c0d0c41e880d4f4145) file).

So we don't need NodeJS environment to test the code, so I was able to remove much of the dependencies / project files.

I did keep compilation step tho. As I was playing with Tetra's codebase, they are using [esbuild](https://esbuild.github.io/) to compile the JS code, so I learnt how to use it. It's a single dependency

Reply:
I'm very happy with this. I'm not super-fond of esbuild, but I'm trusting you on this.